src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -846,6 +849,51 @@ public ListenableFuture<DescribeIndexStatsResponse> describeIndexStats(Struct fi
         return asyncStub.describeIndexStats(describeIndexStatsRequest);
     }
 
+    @Override
+    public ListenableFuture<ListResponse>  list(String namespace) {
+        validateListEndpointParameters(namespace, null, null, null, false, false, false);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setLimit(100).build();
+        return asyncStub.list(listRequest);
+    }
+
+    @Override
+    public ListenableFuture<ListResponse>  list(String namespace, Integer limit) {
+        validateListEndpointParameters(namespace, null, null, limit, false, false, true);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setLimit(limit).build();
+        return asyncStub.list(listRequest);
+    }
+
+    @Override
+    public ListenableFuture<ListResponse> list(String namespace, String prefix) {
+        validateListEndpointParameters(namespace, prefix, null, null, true, false, false);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setPrefix(prefix).setLimit(100).build();
+        return asyncStub.list(listRequest);
+    }
+
+    @Override
+    public ListenableFuture<ListResponse>  list(String namespace, String prefix, Integer limit) {
+        validateListEndpointParameters(namespace, prefix, null, limit, true, false, true);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setPrefix(prefix).
+                setLimit(limit).build();
+        return asyncStub.list(listRequest);
+    }
+
+    @Override
+    public ListenableFuture<ListResponse>  list(String namespace, String prefix, String paginationToken) {
+        validateListEndpointParameters(namespace, prefix, paginationToken, null, true, true, false);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setPrefix(prefix).
+                setPaginationToken(paginationToken).setLimit(100).build();
+        return asyncStub.list(listRequest);
+    }
+
+    @Override
+    public ListenableFuture<ListResponse>  list(String namespace, String prefix, String paginationToken, Integer limit) {
```

Comment:
super nit: I think there's extra space between `ListenableFuture<ListResponse>    list(...` in a lot of these. Is that just me?